internal/proxy/task_upsert.go

```diff
@@ -348,19 +354,59 @@ func (it *upsertTask) PreExecute(ctx context.Context) error {
 		log.Warn("Fail to insertPreExecute", zap.Error(err))
 		return err
 	}
+	it.result.InsertCnt = int64(it.upsertMsg.InsertMsg.NumRows)
 
-	err = it.deletePreExecute(ctx)
+	// query to find if pk were exist
+	// if pk not exist, then skip delete operate
+	queryResult, err := it.queryPK()
 	if err != nil {
-		log.Warn("Fail to deletePreExecute", zap.Error(err))
+		log.Warn("Fail to query pk when upsert", zap.Error(err))
 		return err
 	}
 
-	it.result.DeleteCnt = it.upsertMsg.DeleteMsg.NumRows
-	it.result.InsertCnt = int64(it.upsertMsg.InsertMsg.NumRows)
-	if it.result.DeleteCnt != it.result.InsertCnt {
-		log.Info("DeleteCnt and InsertCnt are not the same when upsert",
-			zap.Int64("DeleteCnt", it.result.DeleteCnt),
-			zap.Int64("InsertCnt", it.result.InsertCnt))
+	err = merr.Error(queryResult.GetStatus())
+	if errors.IsAny(err, merr.ErrCollectionNotLoaded, merr.ErrPartitionNotLoaded,
+		merr.ErrCollectionNotFullyLoaded, merr.ErrPartitionNotFullyLoaded) {
+		it.needDelete = true
+		it.deletePrimaryKeys = it.result.IDs
+		it.upsertMsg.DeleteMsg.NumRows = int64(it.req.NumRows)
+		log.Info("upsert on a unloaded or not loaded fully collection or partition",
+			zap.Error(err), zap.Any("reason", queryResult.GetStatus().Reason))
```

Comment:
the err that converted from status by merr contains the reason, the two log fields would produce dup messages, remove the `reason` field